src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/IDocumentSnapshotExtensions.cs

```diff
@@ -59,4 +59,30 @@ public static bool IsPathCandidateForComponent(this IDocumentSnapshot documentSn
         var fileName = Path.GetFileNameWithoutExtension(documentSnapshot.FilePath);
         return fileName.AsSpan().Equals(path.Span, FilePathComparison.Instance);
     }
+
+    public static Task<RazorCodeDocument> GetFormatterCodeDocumentAsync(this IDocumentSnapshot documentSnapshot)
```

Comment:
Should this take a `CancellationToken` and pass it to `GetDesignTimeDocumentAsync(...)` and then into `project.GetTagHelpersAsync(...)`?

Reply:
A lot of the stack this depends on doesn't currently take cancellation tokens. I think they should so I'll file a tech debt issue.